src/main/java/com/google/sps/servlets/MatchingDateAlgo.java

```diff
@@ -0,0 +1,59 @@
+
+package com.google.sps.servlets;
+
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.cloud.datastore.*;
+import com.google.cloud.datastore.StructuredQuery.PropertyFilter;
+import com.google.gson.Gson;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.HashSet;
+
+@WebServlet("/MatchingAlgoDate")
+public class MatchingDateAlgo extends HttpServlet {
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        //Used to keep track of the good dates. HashSet used for constant "contains" time.
+        HashSet<LongValue> GoodDates = new HashSet<LongValue>();
+
+        //Used to hold dates for each user from datastore
+        List<LongValue> userDays;
+
+        //Used to *temporarily* hold the values which overlap in current GoodDates and userDays
+        HashSet<LongValue> temps = new HashSet<LongValue>();
+
+        //get objects in datastore
+        Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+        String currEvent = request.getParameter("text-input");
+        Query<Entity> query = Query.newEntityQueryBuilder().setKind("Event").setFilter(PropertyFilter.eq("event-id", currEvent)).build();
```

Comment:
I think it would be better if you used the keys from `com.google.sps.Constants`.

Reply:
Also, I think you should be searching for entities with kind `User` right?